src/models/curriculum/unit.ts

```diff
@@ -194,17 +193,15 @@ const populateProblemSections = async (content: Record<string, any>, unitUrl: st
 
 export function getUnitJson(unitId: string | undefined, appConfig: AppConfigModelType) {
   const unitSpec = getUnitSpec(unitId, appConfig);
-  const unitPath = unitSpec?.content;
-  const unitUrl = getAssetUrl(unitPath!);
+  const unitUrl = unitSpec?.content;
   return fetch(unitUrl!)
            .then(async response => {
              if (response.ok) {
                const unitContent = await response.json();
-               const fullUnitContent = unitContent && populateProblemSections(unitContent, unitUrl);
+               const fullUnitContent = unitContent && populateProblemSections(unitContent, unitUrl!);
                return fullUnitContent;
-             }
-             else {
-               throw Error(`Request rejected with status ${response.status}`);
+             } else {
+               return response;
```

Comment:
I've changed this a bit by adding a comment and then being a little more selective about when it returns the response. I'm not sure it's any better, but it may be more clear now.

Currently, in production we return the default unit if the specified unit isn't found. So in `setUnitAndProblem` in src/model/stores, CLUE will get the default unit if the first call to `getUnitJson` returns a status of 404. It behaves similarly with the teacher guide except there's no default for teacher guides, so we simply don't include teacher guide data if the response status is 404.